oracle/blockeligibilityvalidator.go

```diff
@@ -88,38 +79,42 @@ func (v BlockEligibilityValidator) BlockSignedAndEligible(block *types.Block) (b
 	message := serializeVRFMessage(epochBeacon, epochNumber, counter)
 	vrfSig := block.EligibilityProof.Sig
 
-	res, err := v.validateVRF(message, vrfSig, []byte(nodeId.VRFPublicKey))
+	if genesisNoAtx {
+		v.log.Info("skipping VRF validation, genesis block with no ATX") // TODO
+		return true, nil
+	}
+	res, err := v.validateVRF(message, vrfSig, vrfPubkey)
 	if err != nil {
-		v.log.Error("eligibility VRF validation erred: %v", err)
 		return false, fmt.Errorf("eligibility VRF validation failed: %v", err)
 	}
 
 	if !res {
-		v.log.Error("eligibility VRF validation failed")
-		return false, nil
+		return false, fmt.Errorf("eligibility VRF validation failed")
 	}
 
 	eligibleLayer := calcEligibleLayer(epochNumber, v.layersPerEpoch, sha256.Sum256(vrfSig))
 
-	return block.LayerIndex == eligibleLayer, nil
+	if block.LayerIndex != eligibleLayer {
+		return false, fmt.Errorf("block layer (%v) does not match eligibility layer (%v)",
+			block.LayerIndex, eligibleLayer)
+	}
+	return true, nil
 }
 
-func (v BlockEligibilityValidator) getActiveSetSize(block *types.BlockHeader) (uint32, error) {
+func (v BlockEligibilityValidator) getValidAtx(block *types.Block) (*types.ActivationTxHeader, error) {
```

Comment:
1. why the entire block and not only the header?
2. Isn't `getAssATX` a better name for the function?

Reply:
1. The header doesn't have `block.Bytes()` and `block.Sig()` which are required for signature validation.

https://github.com/spacemeshos/go-spacemesh/blob/d084c6d2042c7f67dbe9847d7ea5d0b6e1dd2f54/oracle/blockeligibilityvalidator.go#L114

2. `getValidAtx` may not be a perfect name, but it indicates that it doesn't just *get* the ATX, but also *validates* it. Now that I think about it, I'm considering renaming it `validateSignatureAndAtx` since returning the ATX is actually just a side-effect. The current name doesn't importantly indicate that the method validates the block signature.